scene/3d/skeleton_3d.cpp

```diff
@@ -453,102 +517,121 @@ int Skeleton3D::get_bone_count() const {
 }
 
 void Skeleton3D::set_bone_parent(int p_bone, int p_parent) {
-	ERR_FAIL_INDEX(p_bone, bones.size());
+	const int bone_size = bones.size();
+	ERR_FAIL_INDEX(p_bone, bone_size);
 	ERR_FAIL_COND(p_parent != -1 && (p_parent < 0));
 
-	bones.write[p_bone].parent = p_parent;
+	bones[p_bone].parent = p_parent;
 	process_order_dirty = true;
 	_make_dirty();
 }
 
 void Skeleton3D::unparent_bone_and_rest(int p_bone) {
-	ERR_FAIL_INDEX(p_bone, bones.size());
+	const int bone_size = bones.size();
+	ERR_FAIL_INDEX(p_bone, bone_size);
 
 	_update_process_order();
 
 	int parent = bones[p_bone].parent;
 	while (parent >= 0) {
-		bones.write[p_bone].rest = bones[parent].rest * bones[p_bone].rest;
+		bones[p_bone].rest = bones[parent].rest * bones[p_bone].rest;
 		parent = bones[parent].parent;
 	}
 
-	bones.write[p_bone].parent = -1;
+	bones[p_bone].parent = -1;
 	process_order_dirty = true;
 
 	_make_dirty();
 }
 
 void Skeleton3D::set_bone_disable_rest(int p_bone, bool p_disable) {
-	ERR_FAIL_INDEX(p_bone, bones.size());
-	bones.write[p_bone].disable_rest = p_disable;
+	const int bone_size = bones.size();
+	ERR_FAIL_INDEX(p_bone, bone_size);
+	bones[p_bone].disable_rest = p_disable;
 }
 
 bool Skeleton3D::is_bone_rest_disabled(int p_bone) const {
-	ERR_FAIL_INDEX_V(p_bone, bones.size(), false);
+	const int bone_size = bones.size();
+	ERR_FAIL_INDEX_V(p_bone, bone_size, false);
 	return bones[p_bone].disable_rest;
 }
 
 int Skeleton3D::get_bone_parent(int p_bone) const {
-	ERR_FAIL_INDEX_V(p_bone, bones.size(), -1);
+	const int bone_size = bones.size();
+	ERR_FAIL_INDEX_V(p_bone, bone_size, -1);
 
 	return bones[p_bone].parent;
 }
 
-void Skeleton3D::set_bone_rest(int p_bone, const Transform3D &p_rest) {
-	ERR_FAIL_INDEX(p_bone, bones.size());
+Vector<int> Skeleton3D::get_bone_children(int p_bone) const {
```

Comment:
This function should now return `const LocalVector<int> &`, otherwise you will convert a `LocalVector` (`bones[p_bone].child_bones`) to a `Vector`, which is bad. If you can't convert that, I suggest to replace the `bones[p_bone].child_bones` type and use `Vector` again.

Of course, depending how you solve the issue you have to consider to also change all it's usage:
```c++
// From
Vector<int> child_bones = get_bone_children(p_bone);
// To
const LocalVector<int> &child_bones = get_bone_children(p_bone);
```

Reply:
I converted it back to a `Vector<int>`, as something with exposing the LocalVector properties was causing issues. Here's what the errors looked like:

~~~
[ 86%] Compiling ==> scene/3d/skeleton_ik_3d.cpp
In file included from ./core/object/class_db.h:34,
                 from ./core/config/project_settings.h:34,
                 from ./scene/main/node.h:34,
                 from ./scene/3d/node_3d.h:34,
                 from scene/3d/skeleton_3d.h:36,
                 from scene/3d/skeleton_3d.cpp:31:
./core/object/method_bind.h: In instantiation of 'Variant::Type MethodBindTR<R, P>::_gen_argument_type(int) const [with R = LocalVector<int>; P = {}]':
./core/object/method_bind.h:416:24:   required from here
./core/object/method_bind.h:420:27: error: incomplete type 'GetTypeInfo<LocalVector<int>, void>' used in nested name specifier
  420 |    return GetTypeInfo<R>::VARIANT_TYPE;
      |                           ^~~~~~~~~~~~
./core/object/method_bind.h: In instantiation of 'PropertyInfo MethodBindTR<R, P>::_gen_argument_type_info(int) const [with R = LocalVector<int>; P = {}]':
./core/object/method_bind.h:424:23:   required from here
./core/object/method_bind.h:430:41: error: incomplete type 'GetTypeInfo<LocalVector<int>, void>' used in nested name specifier
  430 |    return GetTypeInfo<R>::get_class_info();
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
./core/object/method_bind.h: In instantiation of 'GodotTypeInfo::Metadata MethodBindTR<R, P>::get_argument_meta(int) const [with R = LocalVector<int>; P = {}]':
./core/object/method_bind.h:440:34:   required from here
./core/object/method_bind.h:444:27: error: incomplete type 'GetTypeInfo<LocalVector<int>, void>' used in nested name specifier
  444 |    return GetTypeInfo<R>::METADATA;
      |                           ^~~~~~~~
./core/object/method_bind.h: In instantiation of 'Variant::Type MethodBindTR<R, P>::_gen_argument_type(int) const [with R = LocalVector<int>; P = {int}]':
./core/object/method_bind.h:416:24:   required from here
./core/object/method_bind.h:420:27: error: incomplete type 'GetTypeInfo<LocalVector<int>, void>' used in nested name specifier
  420 |    return GetTypeInfo<R>::VARIANT_TYPE;
      |                           ^~~~~~~~~~~~
./core/object/method_bind.h: In instantiation of 'PropertyInfo MethodBindTR<R, P>::_gen_argument_type_info(int) const [with R = LocalVector<int>; P = {int}]':
./core/object/method_bind.h:424:23:   required from here
./core/object/method_bind.h:430:41: error: incomplete type 'GetTypeInfo<LocalVector<int>, void>' used in nested name specifier
  430 |    return GetTypeInfo<R>::get_class_info();
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
./core/object/method_bind.h: In instantiation of 'GodotTypeInfo::Metadata MethodBindTR<R, P>::get_argument_meta(int) const [with R = LocalVector<int>; P = {int}]':
./core/object/method_bind.h:440:34:   required from here
./core/object/method_bind.h:444:27: error: incomplete type 'GetTypeInfo<LocalVector<int>, void>' used in nested name specifier
  444 |    return GetTypeInfo<R>::METADATA;
      |                           ^~~~~~~~
In file included from ./core/object/method_bind.h:34,
                 from ./core/object/class_db.h:34,
                 from ./core/config/project_settings.h:34,
                 from ./scene/main/node.h:34,
                 from ./scene/3d/node_3d.h:34,
                 from scene/3d/skeleton_3d.h:36,
                 from scene/3d/skeleton_3d.cpp:31:
./core/variant/binder_common.h: In instantiation of 'void call_with_variant_args_ret_helper(T*, R (T::*)(P ...), const Variant**, Variant&, Callable::CallError&, IndexSequence<Is ...>) [with T = __UnexistingClass; R = LocalVector<int>; P = {}; long unsigned int ...Is = {}]':
./core/variant/binder_common.h:453:35:   required from 'void call_with_variant_args_ret_dv(T*, R (T::*)(P ...), const Variant**, int, Variant&, Callable::CallError&, const Vector<Variant>&) [with T = __UnexistingClass; R = LocalVector<int>; P = {}]'
./core/object/method_bind.h:454:32:   required from 'Variant MethodBindTR<R, P>::call(Object*, const Variant**, int, Callable::CallError&) [with R = LocalVector<int>; P = {}]'
./core/object/method_bind.h:449:18:   required from here
./core/variant/binder_common.h:648:8: error: no match for 'operator=' (operand types are 'Variant' and 'LocalVector<int>')
  648 |  r_ret = (p_instance->*p_method)(VariantCasterAndValidate<P>::cast(p_args, Is, r_error)...);
      |  ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ./core/variant/callable_bind.h:35,
                 from ./core/object/object.h:44,
                 from ./core/variant/binder_common.h:35,
                 from ./core/object/method_bind.h:34,
                 from ./core/object/class_db.h:34,
                 from ./core/config/project_settings.h:34,
                 from ./scene/main/node.h:34,
                 from ./scene/3d/node_3d.h:34,
                 from scene/3d/skeleton_3d.h:36,
                 from scene/3d/skeleton_3d.cpp:31:
./core/variant/variant.h:679:7: note: candidate: 'void Variant::operator=(const Variant&)'
  679 |  void operator=(const Variant &p_variant); // only this is enough for all the other types
      |       ^~~~~~~~
./core/variant/variant.h:679:32: note:   no known conversion for argument 1 from 'LocalVector<int>' to 'const Variant&'
  679 |  void operator=(const Variant &p_variant); // only this is enough for all the other types
      |                 ~~~~~~~~~~~~~~~^~~~~~~~~
In file included from ./core/object/method_bind.h:34,
                 from ./core/object/class_db.h:34,
                 from ./core/config/project_settings.h:34,
                 from ./scene/main/node.h:34,
                 from ./scene/3d/node_3d.h:34,
                 from scene/3d/skeleton_3d.h:36,
                 from scene/3d/skeleton_3d.cpp:31:
./core/variant/binder_common.h: In instantiation of 'void call_with_ptr_args_ret_helper(T*, R (T::*)(P ...), const void**, void*, IndexSequence<Is ...>) [with T = __UnexistingClass; R = LocalVector<int>; P = {}; long unsigned int ...Is = {}]':
./core/variant/binder_common.h:501:43:   required from 'void call_with_ptr_args_ret(T*, R (T::*)(P ...), const void**, void*) [with T = __UnexistingClass; R = LocalVector<int>; P = {}]'
./core/object/method_bind.h:463:40:   required from 'void MethodBindTR<R, P>::ptrcall(Object*, const void**, void*) [with R = LocalVector<int>; P = {}]'
./core/object/method_bind.h:459:15:   required from here
./core/variant/binder_common.h:253:21: error: 'encode' is not a member of 'PtrToArg<LocalVector<int> >'
  253 |  PtrToArg<R>::encode((p_instance->*p_method)(PtrToArg<P>::convert(p_args[Is])...), r_ret);
      |  ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./core/variant/binder_common.h: In instantiation of 'void call_with_variant_args_ret_helper(T*, R (T::*)(P ...), const Variant**, Variant&, Callable::CallError&, IndexSequence<Is ...>) [with T = __UnexistingClass; R = LocalVector<int>; P = {int}; long unsigned int ...Is = {0}]':
./core/variant/binder_common.h:453:35:   required from 'void call_with_variant_args_ret_dv(T*, R (T::*)(P ...), const Variant**, int, Variant&, Callable::CallError&, const Vector<Variant>&) [with T = __UnexistingClass; R = LocalVector<int>; P = {int}]'
./core/object/method_bind.h:454:32:   required from 'Variant MethodBindTR<R, P>::call(Object*, const Variant**, int, Callable::CallError&) [with R = LocalVector<int>; P = {int}]'
./core/object/method_bind.h:449:18:   required from here
./core/variant/binder_common.h:648:8: error: no match for 'operator=' (operand types are 'Variant' and 'LocalVector<int>')
  648 |  r_ret = (p_instance->*p_method)(VariantCasterAndValidate<P>::cast(p_args, Is, r_error)...);
      |  ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ./core/variant/callable_bind.h:35,
                 from ./core/object/object.h:44,
                 from ./core/variant/binder_common.h:35,
                 from ./core/object/method_bind.h:34,
                 from ./core/object/class_db.h:34,
                 from ./core/config/project_settings.h:34,
                 from ./scene/main/node.h:34,
                 from ./scene/3d/node_3d.h:34,
                 from scene/3d/skeleton_3d.h:36,
                 from scene/3d/skeleton_3d.cpp:31:
./core/variant/variant.h:679:7: note: candidate: 'void Variant::operator=(const Variant&)'
  679 |  void operator=(const Variant &p_variant); // only this is enough for all the other types
      |       ^~~~~~~~
./core/variant/variant.h:679:32: note:   no known conversion for argument 1 from 'LocalVector<int>' to 'const Variant&'
  679 |  void operator=(const Variant &p_variant); // only this is enough for all the other types
      |                 ~~~~~~~~~~~~~~~^~~~~~~~~
In file included from ./core/object/method_bind.h:34,
                 from ./core/object/class_db.h:34,
                 from ./core/config/project_settings.h:34,
                 from ./scene/main/node.h:34,
                 from ./scene/3d/node_3d.h:34,
                 from scene/3d/skeleton_3d.h:36,
                 from scene/3d/skeleton_3d.cpp:31:
./core/variant/binder_common.h: In instantiation of 'void call_with_ptr_args_ret_helper(T*, R (T::*)(P ...), const void**, void*, IndexSequence<Is ...>) [with T = __UnexistingClass; R = LocalVector<int>; P = {int}; long unsigned int ...Is = {0}]':
./core/variant/binder_common.h:501:43:   required from 'void call_with_ptr_args_ret(T*, R (T::*)(P ...), const void**, void*) [with T = __UnexistingClass; R = LocalVector<int>; P = {int}]'
./core/object/method_bind.h:463:40:   required from 'void MethodBindTR<R, P>::ptrcall(Object*, const void**, void*) [with R = LocalVector<int>; P = {int}]'
./core/object/method_bind.h:459:15:   required from here
./core/variant/binder_common.h:253:21: error: 'encode' is not a member of 'PtrToArg<LocalVector<int> >'
  253 |  PtrToArg<R>::encode((p_instance->*p_method)(PtrToArg<P>::convert(p_args[Is])...), r_ret);
      |  ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
~~~

Based on the error, I think its a missing function or two that doesn't allow it to work with the properties list, from what I can tell. For now, I figure changing it back to a `Vector<int>` is probably best since I'm not totally sure what would need to be changed to get it working. (Better to not break something accidentally :sweat_smile: )